client/lib/domains/transferred-domains.ts

```diff
@@ -0,0 +1,10 @@
+import { ResponseDomain } from 'calypso/lib/domains/types';
+import { type as domainTypes } from './constants';
+
+export function getTransferredInDomains( domains: ResponseDomain[] ) {
+	return domains.filter( isTransferredInDomain );
+}
+
+export function isTransferredInDomain( domain: ResponseDomain ) {
+	return domain.type === domainTypes.TRANSFER;
+}
```

Comment:
Type guards could also be added here to improve type assertion. However, since this won't be used for now, it's better to filter the list and return the result for simplicity's sake.